src/diffusers/pipelines/kandinsky2_2/pipeline_kandinsky2_2.py

```diff
@@ -243,24 +274,37 @@ def __call__(
                 generator=generator,
             )[0]
 
+            if callback_on_step_end is not None:
```

Comment:
Shouldn't `callback_steps` also work with `callback_on_step_end`?

Reply:
I didn't think `callback_steps` is very meaningful - we could apply any logic we want inside the callback function to skip it no?

Reply:
especially that I've made `timesteps` available for all pipelines

Reply:
IMO it's a slightly nicer experience to not have to implement the timestep/index checking logic for every callback and just have it be executed based on the existing arg.

e.g I want to run a callback every 10 steps and then try it again at every step. I would have to rewrite the callback function

Reply:
probably just update one number in your callback function no? IMO we don't have enough case to justify this parameter
(1) it does not offer additional capability
(2) it is arguably that it may improve efficiency (a little subjective here I think, personally I wouldn't think it's easier that way) - but anyway I don't think there are lot of our users out there want to consistently change the frequency of  their callback function out there so maybe not a factor we need to consider at all

Reply:
Think it's totally fine to just write:

```py
if i % 10:
    # do whatever 
```

in the callback. So agree with @yiyixuxu here